src/connectors/smart-sort/connectSmartSort.ts

```diff
@@ -0,0 +1,130 @@
+import { Connector } from '../../types';
+import { createDocumentationMessageGenerator, noop } from '../../lib/utils';
+
+export type SmartSortConnectorParams = {
+  relevancyStrictness?: number;
+};
+
+type Refine = (relevancyStrictness: number) => void;
+
+export type SmartSortRendererOptions = {
+  isSmartSorted: boolean;
+  refine: Refine;
+};
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'smartSort',
+  connector: true,
+});
+
+export type SmartSortConnector = Connector<
+  SmartSortRendererOptions,
+  SmartSortConnectorParams
+>;
+
+const connectSmartSort: SmartSortConnector = function connectSmartSort(
+  renderFn = noop,
+  unmountFn = noop
+) {
+  return widgetParams => {
+    if (
+      typeof widgetParams.relevancyStrictness !== 'undefined' &&
+      (widgetParams.relevancyStrictness < 0 ||
+        widgetParams.relevancyStrictness > 100)
+    ) {
+      throw new Error(
+        withUsage('The `relevancyStrictness` option must be between 0 and 100.')
+      );
+    }
+
+    type ConnectorState = {
+      refine?: Refine;
+    };
+
+    const connectorState: ConnectorState = {};
+
+    return {
+      $$type: 'ais.smartSort',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const { instantSearchInstance } = renderOptions;
+
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance,
+          },
+          false
+        );
+      },
+
+      dispose({ state }) {
+        unmountFn();
+
+        return state.setQueryParameter('relevancyStrictness', undefined);
+      },
+
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          smartSort: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState({ results, helper }) {
+        if (!connectorState.refine) {
+          connectorState.refine = (relevancyStrictness: number | undefined) => {
+            helper
+              .setQueryParameter('relevancyStrictness', relevancyStrictness)
+              .search();
+          };
+        }
+
+        const { appliedRelevancyStrictness } = results || {};
+
+        return {
+          isSmartSorted:
+            typeof appliedRelevancyStrictness !== 'undefined' &&
+            appliedRelevancyStrictness > 0 &&
+            appliedRelevancyStrictness <= 100,
+          refine: connectorState.refine,
+          widgetParams,
+        };
+      },
+
+      getWidgetSearchParameters(state, { uiState }) {
+        return state.setQueryParameter(
+          'relevancyStrictness',
+          uiState.smartSort?.relevancyStrictness ??
+            state.relevancyStrictness ??
+            widgetParams.relevancyStrictness
```

Comment:
I think by applying the relevancyStrictness by default, we are forcing relevancyStrictness on non-smart sort indices, does that have any undefined behaviour? 

I find it a little odd that this widget would apply the parameter when it shouldn't display.

Possibly this is because we're missing a condition:

normal index: display nothing
smart index (appliedRelevancyStrictness is set):
- appliedRelevancyStrictness === 0: isSmartSorted: false
- appliedRelevancyStrictness !== 0: isSmartSorted: true

what do you think?

Reply:
I had a sync and we're good to remove `relevancyStrictness` from the widget parameter to the component altogether. I had a wrong understanding that it was a part of the spec.

Reply:
In the updated version, I didn't remove the ability to apply relevancyStrictness altogether. I just removed the widget parameter. The component still gets `refine` function and call either `refine(0)` or `refine(undefined)` to toggle the status. https://github.com/algolia/instantsearch.js/pull/4648/commits/2f3409f316f62b53052b3aeeb1e3ee18a67c7573